Update NPM tasks (#2185)
TheSuperHackers/GeneralsGamePatch: Patch104pZH/Design/Tasks/nproject_tasks.txt

 - [MAYBE]       All factions Cargo Planes now have equal stats
 - [IMPROVEMENT][MINOR] Fixed the Cargo Plane propeller texture error that have small thin gray lines
 - [MAYBE]       Standardized unit transport size. Like an Overlord that needs more transport slot than a Technical

>  - [REJECTED][MAJOR] Fixed the issues with firing elevation limit. Now both tanks and artillery can attack very high/low targets alike

Author: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/111

 - [MAYBE]       All factions Cargo Planes now have equal stats
 - [IMPROVEMENT][MINOR] Fixed the Cargo Plane propeller texture error that have small thin gray lines
 - [MAYBE]       Standardized unit transport size. Like an Overlord that needs more transport slot than a Technical

>  - [PENDING][MAJOR] Fixed the issues with firing elevation limit. Now both tanks and artillery can attack very high/low targets alike

Author: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/111

Reviewer: #1050

 Reason: Certain maps have topology which can force units to path around to reach their targets, even if it is within reach of their weapon.

>  See: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/111

Author: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/111

 - [MAYBE] Implemented "Ground Attack" button on all artillery and aircraft units.
 Reason: Allows for more precise force fire than regular force fire. Does not supplant normal force fire but complements it.
 - [DONE] Attackable toxic infantry and burning infantry issue fixed

>  - [DONE][MAJOR] Eternally flying dead aircraft bug (like in 'How To Make A Chinook Bug' video at YouTube) FIXED! You will never encounter

Author: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1018

May need to apply to all vehicles, just in case? Chinook apparently has repo, and PR only covers Aurora, where repro is definite.

Author: Done in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2194

 - [NOTRELEVANT] Dozers clear mines faster. Clearing delay reduced to 50%
 - [MAYBE] All infantry use exploded animation when killed by suicide weapons
 - [NOTRELEVANT] All infantry spill bloods when hitted by bullet and melee weapons!
 - [IMPROVEMENT][MINOR] All buildings with door animation have been standarized so their door opening time are all equal. Some buildings

>  - [DONE][MINOR] Fixed various issues on all infantry model animations, such as running in place or skidding when moving

Author: ```suggestion
 - [IMPROVEMENT][MINOR] Fixed various issues on all infantry model animations, such as running in place or skidding when moving
```

Tracked in https://github.com/TheSuperHackers/GeneralsGamePatch/issues/2204

>  - [DONE][MINOR] All buildings with door animation have been standarized so their door opening time are all equal. Some buildings

Author: Done in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/785 https://github.com/TheSuperHackers/GeneralsGamePatch/pull/786 https://github.com/TheSuperHackers/GeneralsGamePatch/pull/787

   > [NOTRELEVANT] Most vehicles can move backwards correctly
   > [NOTRELEVANT] Helicopters sway a bit when idle
 - [NOTRELEVANT] All artillery weapon shells now have given dust trails like Artillery Barrage shells
 - [DONE][MINOR] Cleaned up many buildings particle effects that were misplaced or attached to invalid locations in the models
 - [IMPROVEMENT][MINOR] Removed the blurry edges on the texture used for the no-build indicator when you try to build structures on invalid places
 - [IMPROVEMENT][MINOR] Parachutes now has housecolors (except crate parachutes)

 EMERGENCY REPAIR:
 - [PENDING] Emergency Repair heal radius increased to 150%
 - [PENDING] Emergency Repair also repair structures

>  - [PENDING] New and better Emergency Repair (this makes the Emergency Repair more valuable as a Generals Powers):

Author: See https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1067

   > [NOTRELEVANT] Most vehicles can move backwards correctly
   > [NOTRELEVANT] Helicopters sway a bit when idle
 - [NOTRELEVANT] All artillery weapon shells now have given dust trails like Artillery Barrage shells

>  - [DONE][MINOR] Cleaned up many buildings particle effects that were misplaced or attached to invalid locations in the models

Author: Various, e.g. https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1373, https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1396, https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1774 Should make a big test at one point to collect remaining issues. I think I went through the buildings once and fixed anything I found already.

Author: ```suggestion
 - [IMPROVEMENT][MINOR] Cleaned up many buildings particle effects that were misplaced or attached to invalid locations in the models
```
undo

 General Balances and Fixes:

 USA DOZER:

>  - [DONE] USA Dozer animation improved. Now Dozers animated when building structures (like when clearing mines). Clearing mines

Author: Dirt fixed in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1019. 

General shovel animation while building is a strange looking gimmick I would not include here. Shovel = disarm, for reabability.

>    animation also fixed (fixing the strange dirt effects)

Author: Dirt fixed in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1019. 

General shovel animation while building is a strange looking gimmick I would not include here. Shovel = disarm, for reabability.

Reviewer: Dirt effect fix is plenty.

   > [IMPROVEMENT][MINOR] The infantry no longer unanimated when performing this ability
 - [NOTRELEVANT] Missile Defender weapon damage increased from 40 to 50
 - [NOTRELEVANT] Missile Defender rate of fire reduced from 1000 to 1600 miliseconds
 - [REJECTED] Added minimum attack range to Missile Defender normal weapon to match Tank Hunter and RPG Trooper weapons (laser lock weapon is unaffected)
 Reason: Missile Defenders will not damage themselves anymore. May affect instances where Rockvees are rammed and they may be unable to retaliate.
 - [NOTRELEVANT] Missile Defender normal attack rockets fuel time rebalanced to match their other factions counterparts (RPG and Tank Hunter)

>  - [DONE] Missile Defender skin changed back to blue from green. This have been applied in original Generals, but removed in

Author: Done in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1179

   > [IMPROVEMENT][MINOR] The infantry no longer unanimated when performing this ability
 - [NOTRELEVANT] Missile Defender weapon damage increased from 40 to 50
 - [NOTRELEVANT] Missile Defender rate of fire reduced from 1000 to 1600 miliseconds

>  - [REJECTED] Added minimum attack range to Missile Defender normal weapon to match Tank Hunter and RPG Trooper weapons (laser lock weapon is unaffected)

Author: Can't find where, but it was rejected, because suiciding into Rockvees is meta, apparently.

Reviewer: The minimum attack range of 0 avoids targeting issues when inside Combat Chinook. Helix + Tank Hunters have this issue.

Author: Maybe none of them should have min attack range.

Reviewer: There is issue report for this:
* #1118

 PILOTS:
 - [DONE] Pilots will not reduce their movement speed when in heroic veterancy
 - [DONE] Pilots never affected by Chemical Suits upgrade anymore due to unfixable bug in the game engine itself
 - [DONE] Pilots no longer can promote Combat Bikes!

 CRUSADER:

>  - [PENDING][MINOR] A single Crusader Tank will no longer create two chassis rubbles

Author: Tracked here: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1736

 CRUSADER:
 - [PENDING][MINOR] A single Crusader Tank will no longer create two chassis rubbles
 - [IMPROVEMENT][MINOR] Promoted Crusader Tank will not create Pilot if died from being crushed, like the other USA vehicles

>  - [REJECTED] Fixed the Composite Armor upgrade effect for Crusader Tanks that gives too many healthpoint bonus

Author: Instead, https://github.com/TheSuperHackers/GeneralsGamePatch/pull/777 was applied

 - [NOTRELEVANT] Ambulance cleanup guard area and scan rate increased, allowing them to clean nearby toxins/radiations better when idle

 SENTRY DRONE:
 - [DONE][MAJOR] Fixed Sentry Drone issue that sometimes not responding to player orders
 - [REJCTED] Sentry Drone can attack while moving
 - [DONE] Sentry Drone stealth detection range increased from 225 to 270
 - [DONE] Sentry Drone no longer immune to toxins
 - [DONE] Sentry Drone build cost reduced from 800 to 650
 - [DONE] Gun upgraded Sentry Drone rate of fire increased from 200 to 130 miliseconds
 - [IMPROVEMENT][MINOR] Fixed the Sentry Drone model UV Mapping and texture errors, also the shadow bugs
 - [DONE] Sentry Drone Guns upgrade research time reduced from 30 to 20 seconds
 - [DONE] Sentry Drone Guns upgrade research cost reduced from 1000 to 800
 - [REJECTED] Now the Drone Armor upgrade also allows them to self repair

>  - [REJECTED] Sentry Drone armor received damage from CRUSH increased from 60% to 80%

Author: See https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1162. Marking some as "DONE" even though changes are slightly different.

 AVENGER:

>  - [DONE][MINOR] Avenger will not block dozer builds anymore!

Author: Can't find PR, but remember a fix, and also can't reproduce on main anymore.

![shot_20230801_175138_1](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/287f33cc-4ab4-4af7-b7fc-96489c0dbeb7)

Reviewer: This #630

 AVENGER:
 - [IMPROVEMENT][MINOR] Avenger will not block dozer builds anymore!
 - [NOTRELEVANT] All Avengers need Strategy Center to be built
 - [DONE] Avenger now use their own armor

>  Reason: Only Airforce General Avenger uses AntiAirArmor, which makes them more vulnerable to JET_MISSILE compared to AvengerArmor. May affect interactions.

Author: Done in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/403

Confusing entry, but that is what is meant as far as I can tell.

 NIGHTHAWK (STEALTH FIGHTER):
 - [NOTRELEVANT] The name officially changed from the Stealth Fighter to Nighthawk

>  - [REJECTED][MINOR] You cannot purchase Bunker Buster upgrade if you haven't unlocked the Nighthawk

Author: Bad change, rejected in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1166

 - [NOTRELEVANT] Spy Drone can be moved and no longer immobile after deployed!

 USA COMMAND CENTER:

>  - [REJECTED] USA Command Center door opening time reduced from 1500 to 1000 miliseconds

Author: China CC was set to 1500 in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/785 
Assuming main branch is as desired now.

Reviewer: I think there was no issue with USA door, so no need to change.

 - [MAYBE] Additional details for USA Supply Center to balance the footprint

 USA WAR FACTORY:

>  - [REJECTED] USA War Factory door opening time reduced from 3250 to 1500 miliseconds

Author: Set to 3300 (strange number...) in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/787

 USA AIRFIELD:
 - [IMPROVEMENT][MINOR] Fixed the missing shadow on USA Airfield model when in damaged conditions

>  - [REJECTED] USA Airfield hangar door opening time reduced from 2000 to 1500 miliseconds

Author: Door times are done now.

Reviewer: Was there any issue with Airfield doors?

Reviewer: I mean issue as in planes clipping through or something like that.

Author: Perhaps, but that is unrelated to the task / npm change. Resolved I'd say.

 - [NOTRELEVANT] Strategy Center no longer reveal surrounding when a battle plan is announced

 SUPPLY DROP ZONE:
 - [DONE][CRITICAL] Supply Drop Zone Cargo Plane are no longer attackable, prevent the annoying "unit lost" report from EVA

>  - [DONE] Supply Drop Zone money now delivered in one crate and will show the money amount upon landing!

Reviewer: Showing the money value is a nice touch. Never thought about that. Black Market and Hacker show money values, so perhaps money crates should too.

>  - [DONE] Supply Drop Zone money now delivered in one crate and will show the money amount upon landing!

Author: ```suggestion
 - [IMPROVEMENT] Supply Drop Zone money now delivered in one crate and will show the money amount upon landing!
    Todo: show the money
```
undo, add note

 PARTICLE CANNON:
 - [NOTRELEVANT] Reduced Particle Cannon build footprint size, so Particle Cannon requires slight less area when placed for build

>  - [DONE][MINOR] Fixed the Particle Cannon animation bug that occurs when the building were disabled with Microwave Tank weapon

Author: Done in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1171

 USA PARADROP:
 - [DONE] The Paradrop special power now deliver the corresponding subfaction's Rangers correctly

>  - [DONE] Paradrop now also deliver some Missile Defenders in the mix:

Author: Done - 40% instead of 20% - in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2026

 - [NOTRELEVANT] AirForce General Supply Center refund value changed from 400 to 450 due to the Chinook price change
 - [NOTRELEVANT] AirForce General Avenger build cost increased from 1600 to 1700
 - [NOTRELEVANT] AirForce General Microwave Tank build cost increased from 800 to 900

>  - [DONE][MINOR] Fixed Airforce General Firebase that use wrong model at snowy-reallydamaged condition

Author: Done in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1177

   > Remove Point Defense Laser ability from AirForce General standard Chinook

 SPECIAL POWERS AIRCRAFTS:

>  - [DONE] Improved AirForce Carpet Bomb efficiency

Author: This can only refer to deploy distance / diagonal movement, which is done in main / will be done via Thyme respectively.

Reviewer: Yes I suspect they tweaked bomb payload deployment timings. We already have this covered for all payload drops.

 SPECIAL POWERS AIRCRAFTS:
 - [DONE] Improved AirForce Carpet Bomb efficiency
 Reason: Depends on what "efficiency" means. Needs to be examined.

>  - [NOTRELEVANT][MAJOR] AirForce Carpet Bomb can be called from other generals Strategy Center

Reviewer: Hmm. When a player is Nuke General and has unlocked the Nuke Bomber but has a China Command Center, then what happens?

>  - [NOTRELEVANT][MAJOR] AirForce Carpet Bomb can be called from other generals Strategy Center

Author: In 1.04? The button for Nuke Bomber appears, but is unusable. Can call normal Bomber from CC via button there.

 COMANCHE:
 - [IMPROVEMENT][MINOR] Correct selection cameo for Stealth Comanche

>  - [NOTRELEVANT] Fixed Stealth Comanche behavior after upgraded with the stealth upgrade. Now they will respond to enemies when idle

Author: This would upset anyone that liked the old behaviour. I think can be rejected.

   > Missile firing rate reduced from 100 to 250 miliseconds, matching the normal Patriot Missile speed
   > Disable duration caused by EMP now varies depending on the affected target health
   > No longer able to one-hit-kill aircrafts

>  - [DONE][MINOR] Fixed EMP Patriot model that use wrong model when constructing or being sold

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1492

Reviewer: Top

 General Balances and Fixes:

 CHINESE DOZER:

>  - [NOTRELEVANT][MINOR] Chinese Dozer animation improved. Now Dozers animated when building structures (like when clearing mines). Clearing mines

Author: See US Dozer. I really dislike this.

Reviewer: Yes I have this. We do not need this.

 GATTLING TANK:
 - [MAYBE] Gattling Tank now can gain horde bonus in group of 5 or more

>  - [DONE][MAJOR] Fixed the bug that makes Gattling Tank keep silent when attacking air

Author: Done in https://github.com/TheSuperHackers/GeneralsGamePatch/pulls/293

 TROOP CRAWLER:
 - [NOTRELEVANT] You need a Chinese Barracks to build Troop Crawlers

>  - [DONE][MINOR] Better Troop Crawler stealth detection animations

Author: Done in #350

 - [DONE][MINOR] Better Listening Outpost stealth detection animations
 - [NOTRELEVANT] Changed Listening Outpost armor from TankArmor to TruckArmor (also applied to Attack Outpost)
 - [NOTRELEVANT] Listening Outpost now will not drop salvage crates when destroyed

>  - [DONE][MINOR] Fixed various issues with Listening Outpost model

Author: Fixed bugged smoke in #1774. Dunno what else there would be.

Author: Extra fire bones (3 on four front sides). No doors on backside, instead, the barrels emerge from the corners of the building (strange). Not nearly enough ports on Infa Bunker for 10 guys.

Author: Agree. For purposes of this list, would say resolved though.

Author: I definitely clicked "both sides" on those when exporting the model, so if that was a bug in 1.04, I think I fixed it by accident lol

Author: Headlights now fixed. I think this task is done.

 - [DONE][MINOR] Fixed various issues with Listening Outpost model

 OVERLORD:

>  - [DONE][MAJOR] Upgraded Overlords will not block dozer builds anymore!

Author: Done in #630

 - [NOTRELEVANT] Nuke Cannon attack range increased from 350 to 420
 - [IMPROVEMENT][MINOR] Nuke Cannon barrel will recoil properly when attacking with Nuclear Shells
 - [NOTRELEVANT] Nuke Cannon no longer crushable by Overlord tanks

>  - [DONE][MINOR] Fixed Nuke Cannon weapon shells explosion effect that bit off from the center of the explosion

Author: Done in #1122

Reviewer: This applies to all sorts of nuke explosions. A code fix would be better than the workarounds. But is not important.

Author: Agree. For purposes of this list, would say resolved though.

 - [IMPROVEMENT][MINOR] Nuke Cannon barrel will recoil properly when attacking with Nuclear Shells
 - [NOTRELEVANT] Nuke Cannon no longer crushable by Overlord tanks
 - [IMPROVEMENT][MINOR] Fixed Nuke Cannon weapon shells explosion effect that bit off from the center of the explosion

>  - [REJECTED][MINOR] You cannot purchase Neutron Shells upgrade if you haven't unlocked the Nuke Cannon

Author: Bad change, rejected in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1166

Reviewer: Reject

 NUCLEAR REACTOR:
 - [MAYBE] All Nuclear Reactors will not die because of their Overcharge ability. Overcharge ability will automatically turned off
   on reactor's low healthpoint

>  - [DONE][MINOR] Optimized all Nuclear Reactor model codes. Damaged states smokes and fires appears at right positions

Author: Done in #1395

 BUNKER:
 - [DONE] Fixed the issue of Chinese Bunkers that not appeared on the radar

>  - [DONE][MAJOR] Fixed Chinese Bunker fireport issue where garrisons sometimes attacking from the center bottom of the Bunker

Author: Done in #2133

Reviewer: I wonder how they fixed it. Same method or perhaps they found another trick to avoid shooting from 0,0,0?

 - [NOTRELEVANT] Now you cannot build multiple Internet Centers from different subfactions
 - [IMPROVEMENT][MINOR] New constructing animation for Internet Center. Now it have unique scaffolding (not using the Supply Center one)
 - Some changes on Satellite Hack upgrades due to new Spy Plane ability (see below):
   > [NOTRELEVANT] Satellite Hack I reveal now non-permanent. Now it only reveal enemy Command Centers for 25 seconds each 4 minutes.

>    > [DONE][MAJOR] Also fixed Satellite Hack I bug that sometimes not revealing enemy newly built Command Centers (hopefully)

Author: First one not relevant.
Second done in #791

 - [NOTRELEVANT] Neutron Mines now exclusive for Nuke General! The other Chinese generals will get the new EMP Mines instead. (see below)

 GATTLING WEAPONS:

>  - [DONE][MAJOR] Gattling weapons now properly show the muzzle flash when attacking ground units

Author: Done in #1341

 - [NOTRELEVANT] RPG Trooper rate of fire reduced from 1000 to 1600 miliseconds

 TERRORIST:

>  - [DONE][MINOR] Terrorist now use correct death animations (including exploded and burned animations)

Author: Pretty sure you fixed this.

Author: ```suggestion
 - [PENDING][2084] Terrorist now use correct death animations (including exploded and burned animations)
```

>  - [REJECTED] Now you can manually detonate the Terrorist

Author: Demolitions button was rejected for Terrorist, so I think this is out of question.

 - [NOTRELEVANT] You require a Palace instead an Arms Dealer to train the Saboteur
 - [MAYBE] Saboteur now steal $2000 from supply centers, make them more valueable.
 - [IMPROVEMENT][MINOR] Fully new animation set for Saboteur! Now they're not sharing same movement style with Workers anymore.

>  - [NOTRELEVANT] Saboteur now will be revealed when taking damage

Author: No need to nerf these niche units. Other units disappear under fire too and that's ok

Reviewer: If Saboteur is no single outlier then this needs no looking yes.

Author: TAKING_DAMAGE flag in StealthForbiddenConditions value of StealthUpdate module.

Note that all units have StealthUpdate via Default/Object.ini unless overridden in the object ("OverrideableByLikeKind").

No TAKING_DAMAGE in there, so the vast amount of units are not revealed while taking damage). TAKING_DAMAGE is mostly (exclusively?) used on buildings.

Reviewer: There was this change which supposedly streamlined behaviour:

* #1706

Author: Yeah, forgot about this Stealthgen Rebel is the lone exception in 1.04, for no reason.

 - [IMPROVEMENT][MAJOR] Mob members movement speed increased a bit and will not reduced when injured, so they can catch the squad leader faster

 HIJACKER:

>  - [IMPROVEMENT] Hijacker can no longer hijack drone units like Sentry Drones

Author: This is a good change imo. 

- (Irrelevant) buff to Sentry Drone
- Sentry Drone is meant to counter spec ops units like this. For example, it is immune to Jarmen Kell snipe, and it can detect hidden units.
- Logically, Hijacker is meant to knife the pilot and take over the vehicle. That wouldn't work with a remote controlled / ai controlled Drone
- Sentry Drone cannot vet up, so Hijacker cannot emerge from it when destroyed like other USA units (no Pilot logic)

Reviewer: Yes this makes sense. Sentry Drone has no Pilot, so Hijacker should not be able to steal it. I did not even know it can, because no one uses Sentry Drones to begin with :)

Author: ![shot_20230802_230950_1](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/e0186051-4dbb-4438-bf6a-14218aea00c1)

Cannot reproduce, Sentry Drone is not capturable by Hijacker.

Author: Remving DRONE kindof from SentryDrone makes it capturable. 

I conclude that: 

- `ConvertToHijackedVehicleCrateCollide` `ForbiddenKindOf` defaults to `DRONE` and maybe more. 
- By trying to make Combat Bikes no hijackable, NPM authors added `ForbiddenKindOf` `CLIFF_JUMPER` to the module
- thus overwriting the default `DRONE` `ForbiddenKindOf`
- thus introducing the bug that later was fixed again.

We fixed Hijacker vs Combat Bike differently: by adding IMMUNE_TO_CAPTURE to Combat Bikes.

Nothing to do here.
```suggestion
 - [NOTRELEVANT] Hijacker can no longer hijack drone units like Sentry Drones
```

Author: Resolved.

>  - [DONE] Hijacker can no longer hijack Combat Bikes

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/879

 - [NOTRELEVANT] Hijacker now will be able to respawn correctly from all vehicles, not just from USA vehicles! (some exceptions apply)
 - [NOTRELEVANT] Hijacker build cost increased from 400 to 600
 - [NOTRELEVANT] Hijacker build time increased from 10 to 12 seconds

>  - [NOTRELEVANT] Hijacker now will be revealed when taking damage (same goes to Stealth General Hijacker)

Author: See Sabo

 QUAD CANNON:
 - Various changes for Quad Cannon weapons for more consistent growth and balance:

>    > [NOTRELEVANT] Changed Quad Cannon weapon damage type from SMALL_ARMS to GATTLING

Author: This would reduce damage vs buildings by -80% and make it no longer be able to clear Stinger Sites. I think a drastic change like this can be outright rejected.

   > [NOTRELEVANT] 2nd salvage upgrade weapon damage increased from 8 to 12
   > [NOTRELEVANT] 2nd salvage upgrade weapon damage against air targets increased from 5 to 10
   > [NOTRELEVANT] 2nd salvage upgrade weapon rate of fire reduced from 25 to 100 miliseconds

>  - [DONE][MINOR] Quad Cannon now show the muzzle flash when attacking ground units

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1340

 - [NOTRELEVANT] Quad Cannon armor received damage from COMANCHE_VULCAN reduced from 50% to 40%

 TOXIN TRACTOR:

>  - [DONE] Toxin Tractor no longer immune to radiations

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2052

 TOXIN TRACTOR:
 - [DONE] Toxin Tractor no longer immune to radiations
 - [NOTRELEVANT] Toxin Tractor now little more resistant to infantry and bullet weapons

>  - [DONE][MINOR] Fixed the Toxin Tractor salvage upgrade models that didn't show/hide correct toxin tanks

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/820

   > received damage from JET_MISSILES reduced from 100% to 50%
   > received damage from STEALTHJET_MISSILES increased from 150% to 250%
   > received damage from AURORA_BOMB reduced from 250% to 200%

>  - [NOTRELEVANT][MINOR] Fixed Tunnel Network turret model that use wrong texture on snow conditions

Author: can't reproduce. Correct texture, day and night

![shot_20230803_224028_2](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/a5ca9555-5d52-403a-afde-918164ab892c)

![shot_20230803_224841_1](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/a4d4a6d5-1dfe-43eb-9b93-3130f67dc352)

Where's the issue?

Reviewer: Perhaps another sub faction tunnel?

Author: ![shot_20230804_074124_2](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/76651d7b-4e74-4221-b0ec-c937a732ac4c)

![shot_20230804_073756_1](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/6be2d9b5-16dd-4c34-bbe1-c2bfe5433433)

Tox Tunnel uses Toxin Truck textures for turret, and as a vehicle, that doesn't have a snow version (or a really damaged version). I don't think changing that is necessary though. lgtm

Author: Sneak Attack is its own can of worms, with a very choppy transition between emerging tunnel and usable tunnel.

Author: Resolved

 - [NOTRELEVANT] Tunnel Network no longer block unit sights, allowing other units to attack enemies behind

 STINGER SITE:

>  - [DONE] Stinger Site can be attacked by Ranger Flash Bangs again

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/297

"again" refers to the fact that, as of 1.08, this bug is not present in CCG.

![shot_20230801_183048_1](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/bda29a75-c0bb-4100-b93c-97471fca00b4)

 - [MAYBE] Stinger Site can have Fortified Structures upgrade
 - [NOTRELEVANT] Stinger Site no longer block unit sights, allowing other units to attack enemies behind
 - [MAYBE] Stinger Soldiers now can properly take damage from radiations like the other normal infantry

>  - [NOTRELEVANT] Individual Stinger Soldier can't be targeted by Burton knife attacks anymore

Author: Can't reproduce. Maybe refers to bug introduced by NPM and later resolved?

 - [DONE] Added the missing Stop button in Palace command bar
 - [NOTRELEVANT] Palace build cost increased from 2500 to 2800
 - [NOTRELEVANT] Palace healthpoint reduced from 3000 to 2500

>  - [DONE][MAJOR] Fixed the Palace model that lost some fireports after upgraded with Fortified Structures

Author: Done in https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1097

 - [DONE] SCUD Storm's SCUD-bug fixed! You can no longer use the infamous exploit to launch SCUD Storm repeatedly ever again!
 - [MAYBE][MINOR][GITHUB] New SCUD Storm animation! Now the missiles will be hidden underground until they are ready to fire.
 - [DONE] SCUD Storm spawn corresponding toxin field level when destroyed

>  - [NOTRELEVANT] SCUD Storm can have Fortified Structures upgrade

Author: This would drastically affect the SW spam meta. I think only bad things can come of this.

Reviewer: Yes this is not good. Making it harder to kill would just make GLA unnecessarily better in late game again.

 - [MAYBE][MINOR][GITHUB] New SCUD Storm animation! Now the missiles will be hidden underground until they are ready to fire.
 - [DONE] SCUD Storm spawn corresponding toxin field level when destroyed
 - [MAYBE] SCUD Storm can have Fortified Structures upgrade

>  - [REJECTED] SCUD Storm missiles can create salvage crate from enemy kills

Author: ```suggestion
 - [MAYBE] SCUD Storm missiles can create salvage crate from enemy kills
```

 - [DONE] Fixed the bug where SCUD Storm being immune to the damage caused by its own launched missiles

 GLA HOLES:

>  - [NOTRELEVANT][MINOR] GLA Holes now spawn the corresponding subfaction's Worker

Author: Unnecessary change I think, Workers are identical anyway. Except is GC Tox, where Worker is immune to poison. GC Tox Worker is used in GC Tox buildings. Too lazy to find the PR that added this to main. May have also already been the case in 1.04.  

Command Set can be seen with a trick, but not used. 

![shot_20230801_183822_3](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/c811b61f-9ddb-4783-8fa0-6949a9a076b0)

Maybe add a dummy Worker without/ with empty command set? Would need .SCB support, so dummy worker appears in priority list for AI.

Reviewer: There is merit to this change to properly support different sub faction skins. So if a mod was to add different worker skins, then this fix would be good to have.

Author: That makes sense. Consider it done.

```suggestion
 - [DONE][MINOR] GLA Holes now spawn the corresponding subfaction's Worker
```

Author: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2187

 TOXIN SHELLS:
 - [DONE] Toxin field area of effect will not reduced after Anthrax upgrade purchased

>  - [NOTRELEVANT] Toxin Shells toxin field effect no longer harm structures (like how radiation field works)

Author: Can we discard this? Toxin Tractor, Toxin Tunnel and Toxin Rebel can all damage buildings. Toxin alias for all sorts of chemical and acid weapons. I think current state is fine.

Author: Acid wears down buildings, radiation, only in cosmic quantities. Idk.

Author: ```suggestion
 - [MAYBE] Toxin Shells toxin field effect no longer harm structures (like how radiation field works)
```
undo

   the Sneak Attack finished deploying.

 SPECIAL POWERS:

>  - [DONE] Anthrax Bomb generals power starts with green toxin field instead the blue one, and it can be upgraded with Anthrax

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/826

 SPECIAL POWERS:
 - [DONE] Anthrax Bomb generals power starts with green toxin field instead the blue one, and it can be upgraded with Anthrax
   Beta/Gamma upgrade

>  - [DONE] Anthrax Bomb toxin damage per second rebalanced and do toxin damage from the start

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2044 and https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2045

Reviewer: The changes are not yet merged, so perhaps should not yet be marked done.

Author: ```suggestion
 - [PENDING][2044,2055] Anthrax Bomb toxin damage per second rebalanced and do toxin damage from the start
```

 - [NOTRELEVANT] Toxin Shells upgrade no longer affects Demolition General Marauder

 ADVANCED DEMO TRAP:

>  - [DONE][MINOR] Fixed the wrong model of Advanced Demo Trap when under construction

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/563

 DEMOLITIONS UPGRADE:
 - [MAYBE] Improved Demolition ability
 Reason: Vague. May include the fixed below, may be referring to higher damage or radius.

>  - [DONE][MINOR] Fixing Demolition General infantry and vehicles suicide death animations

Author: Done in: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1722 and others. Afaik, they all work fine now.

Reviewer: Vehicles are correct I think. Infantry I am not sure.

Author: I think after https://github.com/TheSuperHackers/GeneralsGamePatch/pull/101 demo infantry should be good.

Author: Lgtm from quick test. Resolved.
